projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/KubernetesService.java

```diff
@@ -370,43 +448,136 @@ public void deleteNamespace(String namespaceName) throws ApiException {
+        } catch (ApiException e) {
+            log.warn("Failed to read k8s cron jobs: ", e);
+        }
+
+        return cronJobs == null ? Collections.emptyList() : cronJobs.getItems().stream()
+                .map(cronJob -> mapV1beta1CronJobToDeploymentStatus(cronJob, null))
+                .flatMap(Optional::stream)
+                .collect(Collectors.toList());
+    }
+
+   public List<JobDeploymentStatus> readV1CronJobDeploymentStatuses() {
       log.debug("Reading all k8s cron jobs");
-      V1beta1CronJobList cronJobs = null;
+      V1CronJobList cronJobs = null;
       try {
-         cronJobs = new BatchV1beta1Api(client).listNamespacedCronJob(namespace, null, null, null, null, null, null, null, null, null, null);
+         cronJobs = new BatchV1Api(client).listNamespacedCronJob(namespace, null, null, null, null, null, null, null, null, null, null);
       } catch (ApiException e) {
          log.warn("Failed to read k8s cron jobs: ", e);
       }
 
       return cronJobs == null ? Collections.emptyList() : cronJobs.getItems().stream()
-            .map(cronJob -> mapCronJobToDeploymentStatus(cronJob, null))
+            .map(cronJob -> mapV1CronJobToDeploymentStatus(cronJob, null))
             .flatMap(Optional::stream)
             .collect(Collectors.toList());
    }
 
     public void startNewCronJobExecution(String cronJobName, String executionId, Map<String, String> annotations,
                                          Map<String, String> envs, Map<String, Object> extraJobArguments, String jobName) throws ApiException {
-        var cron = initBatchV1beta1Api().readNamespacedCronJob(cronJobName, namespace, null);
+        if(getK8sSupportsV1CronJob()) {
+            startNewV1CronJobExecution(
+                    cronJobName,
+                    executionId,
+                    annotations,
+                    envs,
+                    extraJobArguments,
+                    jobName);
+        } else {
+            startNewV1beta1CronJobExecution(
+                    cronJobName,
+                    executionId,
+                    annotations,
+                    envs,
+                    extraJobArguments,
+                    jobName);
+        }
+    }
 
+    private void startNewV1beta1CronJobExecution(String cronJobName, String executionId, Map<String, String> annotations,
+                                         Map<String, String> envs, Map<String, Object> extraJobArguments, String jobName) throws ApiException {
+        var cron = initBatchV1beta1Api().readNamespacedCronJob(cronJobName, namespace, null);
         Optional<V1beta1JobTemplateSpec> jobTemplateSpec = Optional.ofNullable(cron)
-              .map(v1beta1CronJob -> v1beta1CronJob.getSpec())
-              .map(v1beta1CronJobSpec -> v1beta1CronJobSpec.getJobTemplate());
+                .map(v1beta1CronJob -> v1beta1CronJob.getSpec())
```

Comment:
all of those lambdas could be optimised to direct method reference, for example
`v1beta1CronJob -> v1beta1CronJob.getSpec()` becomes `V1beta1CronJob::getSpec`

Reply:
noted